themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/applications-page/ApplicationsPage.tsx

```diff
@@ -15,21 +15,196 @@
+    super(props);
+    this.state = {
+      isRowOpen: [],
+      applications: []
+    };
+
+    this.fetchApplications();
+  }
+
+  private removeConsent = (clientId: string) => {
+    AccountServiceClient.Instance.doDelete("/applications/" + clientId + "/consent")
+      .then(() => {
+        this.fetchApplications();
+      });
+  }
+
+  private onToggle = (row: number): void => {
+    const newIsRowOpen: boolean[] = this.state.isRowOpen;
+    newIsRowOpen[row] = !newIsRowOpen[row];
+    this.setState({ isRowOpen: newIsRowOpen });
+  };
+
+  private fetchApplications(): void {
+    AccountServiceClient.Instance.doGet("/applications")
+      .then((response: AxiosResponse<Object>) => {
+        const applications = response.data as Application[];
+        this.setState({
+          isRowOpen: this.collapseRows(applications),
+          applications: applications
+        });
+      });
+  }
+
+  private collapseRows(applications: Application[]): boolean[] {
+    const openRows: boolean[] = new Array<boolean>().fill(true);
+    applications.forEach((application: Application, appIndex: number) => {
+      openRows[appIndex] = false;
+    });
+    return openRows;
+  }
+
+  public render(): React.ReactNode {
+    return (
+      <ContentPage title="applications">
+        <DataList id="applications-list" aria-label="Applications">
+          {this.state.applications.map((application: Application, appIndex: number) => {
+            return (
+              <DataListItem key={'application-' + appIndex} aria-labelledby="applications-list" isExpanded={this.state.isRowOpen[appIndex]}>
+                <DataListItemRow>
+                  <DataListToggle
+                    onClick={() => this.onToggle(appIndex)}
+                    isExpanded={this.state.isRowOpen[appIndex]}
+                    id={'applicationToggle' + appIndex}
+                    aria-controls="expandable"
+                  />
+                  <DataListItemCells
+                    dataListCells={[
+                      <DataListCell width={2} key={'app-' + appIndex}>
+                        <BuilderImageIcon /> <Text component={TextVariants.small}>{application.clientId}</Text>
+                      </DataListCell>,
+                      <DataListCell width={2} key={'internal-' + appIndex}>
+                        <span>{application.inUse && !application.userConsentRequired ? 'Internal' : 'Third-party'}
+                          {application.offlineAccess ? ', Offline Access' : ''}
+                        </span>
+                      </DataListCell>,
+                      <DataListCell width={2} key={'status-' + appIndex}>
+                        <span>{application.inUse ? 'In use' : 'Not in use'}</span>
+                      </DataListCell>,
+                      <DataListCell width={4} key={'baseUrl-' + appIndex}>
+                        <a href={application.baseUrl}><LinkIcon /> {application.baseUrl}</a>
```

Comment:
Just a suggestion. Should the link be open in a new tab (`target="_blank"`) similarly to Admin Console. It's really convenient. :)

Reply:
@vmuzikar absolutely np about adding this. @ssilvert what do you think?

Reply:
Sounds good to me.